api/main.py

```diff
@@ -40,22 +40,37 @@ async def pubsub_startup():
     pubsub = await PubSub.create()
 
 
-async def get_current_user(token: str = Depends(oauth2_scheme)):
+async def get_current_user(
+        security_scopes: SecurityScopes,
+        token: str = Depends(oauth2_scheme)):
     """Return the user if authenticated successfully based on the provided
     token"""
+
+    if security_scopes.scopes:
+        authenticate_value = f'Bearer scope="{security_scopes.scope_str}"'
+    else:
+        authenticate_value = "Bearer"
+
     if token == 'None':
         raise HTTPException(
             status_code=status.HTTP_401_UNAUTHORIZED,
             detail="Missing credentials",
-            headers={"WWW-Authenticate": "Bearer"},
+            headers={"WWW-Authenticate": authenticate_value},
         )
-    user = await auth.get_current_user(token)
+    user, token_scopes = await auth.get_current_user(token)
     if user is None:
         raise HTTPException(
             status_code=status.HTTP_401_UNAUTHORIZED,
             detail="Could not validate credentials",
-            headers={"WWW-Authenticate": "Bearer"},
+            headers={"WWW-Authenticate": authenticate_value},
         )
+    for scope in security_scopes.scopes:
+        if scope not in token_scopes:
+            raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail="Access denied",
+                headers={"WWW-Authenticate": authenticate_value},
+            )
```

Comment:
Similarly, the `get_current_user()` method could be checking the scopes and return a `User` object only if there is a matching one that meets the scopes criteria rather than doing the check on the caller's side.